src/annotations/components/controlBar/AnnotationsControlBar.tsx

```diff
@@ -1,35 +1,48 @@
 // Libraries
-import React, {FC} from 'react'
-import {useSelector} from 'react-redux'
+import React, {useState, FC} from 'react'
+import {useDispatch, useSelector} from 'react-redux'
 import {useHistory, useParams} from 'react-router-dom'
 
 // Components
 import {
+  ComponentColor,
   ComponentSize,
   FlexBox,
   FlexBoxChild,
   IconFont,
+  InputLabel,
+  InputToggleType,
   JustifyContent,
   SquareButton,
+  Toggle,
 } from '@influxdata/clockface'
 import ErrorBoundary from 'src/shared/components/ErrorBoundary'
 import Toolbar from 'src/shared/components/toolbar/Toolbar'
+import {toggleSingleClickAnnotations} from 'src/annotations/actions/creators'
 import {AnnotationPills} from 'src/annotations/components/controlBar/AnnotationPills'
 import {AnnotationsSearchBar} from 'src/annotations/components/controlBar/AnnotationsSearchBar'
 
 // Selectors
-import {getAnnotationControlsVisibility} from 'src/annotations/selectors'
+import {
+  getAnnotationControlsVisibility,
+  isSingleClickAnnotationsEnabled,
+} from 'src/annotations/selectors'
 
 // Constants
 import {ORGS, SETTINGS, ANNOTATIONS} from 'src/shared/constants/routes'
 
 export const AnnotationsControlBar: FC = () => {
   const history = useHistory()
-  const isVisible = useSelector(getAnnotationControlsVisibility)
   const {orgID} = useParams<{orgID: string; dashboardID: string}>()
 
-  if (!isVisible) {
-    return null
+  const inWriteMode = useSelector(isSingleClickAnnotationsEnabled)
+  const [annotationWriteMode, setAnnotationWriteMode] = useState(inWriteMode)
+
+  const dispatch = useDispatch()
+
+  const onModeChange = () => {
```

Comment:
naming nit: this is a function that makes a change to state. the fact that it happens to be used as a handler shouldn't affect the name of it. i think it's more powerful to name the function for what it does, rather than how it might be used.

something like `changeMode` vs `onModeChange`. 

then in the component, you'd have something like

`onChange={changeMode}`